Source/Scene/ParticleSystem.js

```diff
@@ -0,0 +1,539 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/Cartesian2',
+        '../Core/Cartesian3',
+        '../Core/Event',
+        '../Core/Matrix4',
+        '../Core/Math',
+        '../Core/JulianDate',
+        '../Core/Color',
+        './BillboardCollection',
+        './Particle',
+        './CircleEmitter'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        Cartesian2,
+        Cartesian3,
+        Event,
+        Matrix4,
+        CesiumMath,
+        JulianDate,
+        Color,
+        BillboardCollection,
+        Particle,
+        CircleEmitter) {
+    "use strict";
+
+
+    // An array of available particles that we can reuse instead of allocating new.
+    var particlePool = [];
+
+    /**
+     * Gets a Particle to add to the particle system, reusing Particles from the pool if possible.
+     */
+    function getOrCreateParticle() {
```

Comment:
Are you sure memory management this simple is OK?  E.g., we don't need to time stamp particles in `particlePool` and free over time?

Have you also tried multiple particle systems in the same scene?